FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/MainSignInFragmentEnterpriseFlowTest.kt

```diff
@@ -94,23 +97,50 @@ class MainSignInFragmentEnterpriseFlowTest : BaseSignTest() {
           }
 
           request.requestUrl?.encodedPath == "/gmail/v1/users/me/messages"
-              && request.requestUrl?.queryParameter("q")
-            ?.startsWith("from:${EMAIL_FES_ENFORCE_ATTESTER_SUBMIT}") == true ->
-            return MockResponse()
-              .setResponseCode(HttpURLConnection.HTTP_OK)
-              .setBody(
-                ListMessagesResponse().apply {
-                  factory = GsonFactory.getDefaultInstance()
-                  messages = emptyList()
-                  resultSizeEstimate = 0
-                }.toString()
-              )
+              && request.requestUrl?.queryParameterNames?.contains("q") == true -> {
+            val q = requireNotNull(request.requestUrl?.queryParameter("q"))
+            return when {
+              q.startsWith("from:${EMAIL_FES_ENFORCE_ATTESTER_SUBMIT}") -> MockResponse()
+                .setResponseCode(HttpURLConnection.HTTP_OK)
+                .setBody(
+                  ListMessagesResponse().apply {
+                    factory = GsonFactory.getDefaultInstance()
+                    messages = emptyList()
+                    resultSizeEstimate = 0
+                  }.toString()
+                )
+
+              q.startsWith("from:${EMAIL_GMAIL}") ->
+                prepareMockResponseForPublicDomains(EMAIL_GMAIL)
+
+              q.startsWith("from:${EMAIL_GOOGLEMAIL}") ->
+                prepareMockResponseForPublicDomains(EMAIL_GOOGLEMAIL)
+
+              else -> MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
+            }
+          }
         }
 
         return MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
       }
     })
 
+  private fun prepareMockResponseForPublicDomains(string: String) =
+    MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
+      .setHeader("Content-Type", Json.MEDIA_TYPE)
+      .setBody(GoogleJsonErrorContainer().apply {
+        factory = GsonFactory.getDefaultInstance()
+        error = GoogleJsonError().apply {
+          code = HttpURLConnection.HTTP_NOT_FOUND
+          message = string
+          errors = listOf(GoogleJsonError.ErrorInfo().apply {
```

Comment:
Does this mean mock server will just return email value as error string(with 404 error)?
Not sure if this is same behavior in live one but I think email error string isn't just enough?

Reply:
I'm testing the following case

![temp](https://user-images.githubusercontent.com/2863246/213642620-c52a8d07-6f5f-4f39-8d81-05f4b5423761.png)

`testFlowForPublicEmailDomainsGmail` and `testFlowForPublicEmailDomainsGoogleMail` check that for public domains like `gmail.com` and `googlemail.com` we will use the logic `login as a regular user`. This login includes the following steps:

1. Sign-in via `Continue with Gmail`
2. Find backups via `Gmail API`.

`prepareMockResponseForPublicDomains` returns an error for `testFlowForPublicEmailDomainsGmail` and `testFlowForPublicEmailDomainsGoogleMail`(for step 2). In this case, it will mean that we use the right logic and execute the right requests. It doesn't matter what to return here. It's just for identification.

Reply:
Got it. Makes sense